siad/renter.go

```diff
@@ -33,6 +41,22 @@ func (d *daemon) renterDownloadHandler(w http.ResponseWriter, req *http.Request)
 	writeSuccess(w)
 }
 
+// renterDownloadqueueHandler handles the API call to request the download
+// queue.
+func (d *daemon) renterDownloadqueueHandler(w http.ResponseWriter, req *http.Request) {
+	downloads := d.renter.DownloadQueue()
+	downloadSet := make([]DownloadInfo, 0, len(downloads))
+	for _, dl := range downloads {
+		downloadSet = append(downloadSet, DownloadInfo{
+			Completed:   dl.Completed(),
+			Destination: dl.Destination(),
+			Nickname:    dl.Nickname(),
+		})
```

Comment:
A few reasons. First, that's how it works for file info. Second, I like having interfaces as the modules package objects instead of having structs. Third, it gives the module flexibility to change the getters and not adhere to the struct.

For example, destination could go to a download folder, and the path seen by the renter could be a relative path. Then, when Destination() is called, it can append the relative path to the download path. This gives you flexibility to change the download folder without needing to modify all of your active downloads.

It's not a perfect example. Other more likely examples include adding a field that gives some indication of progress for the download.